wetterdienst/indexing/file_index_creation.py

```diff
@@ -1,69 +1,120 @@
+    DWDCDCDataPath,
 )
-from wetterdienst.constants.metadata import ARCHIVE_FORMAT, STATION_ID_REGEX
+from wetterdienst.constants.metadata import ArchiveFormat, STATION_ID_REGEX, RADOLAN_HISTORICAL_DT_REGEX, \
+    RADOLAN_RECENT_DT_REGEX
 from wetterdienst.enumerations.column_names_enumeration import DWDMetaColumns
+from wetterdienst.enumerations.datetime_format_enumeration import DatetimeFormat
 from wetterdienst.enumerations.parameter_enumeration import Parameter
 from wetterdienst.enumerations.period_type_enumeration import PeriodType
 from wetterdienst.enumerations.time_resolution_enumeration import TimeResolution
 from wetterdienst.file_path_handling.path_handling import (
     build_path_to_parameter,
-    list_files_of_climate_observations,
+    list_files_of_dwd,
 )
 
 
-@functools.lru_cache(maxsize=None)
-def create_file_index_for_dwd_server(
-    parameter: Parameter, time_resolution: TimeResolution, period_type: PeriodType
+@lru_cache
+def create_file_index_for_climate_observations(
+        parameter: Parameter,
+        time_resolution: TimeResolution,
+        period_type: PeriodType
 ) -> pd.DataFrame:
-    """
-    Function (cached) to create a file index of the DWD station data. The file index
-    is created for an individual set of parameters.
-    Args:
-        parameter: parameter of Parameter enumeration
-        time_resolution: time resolution of TimeResolution enumeration
-        period_type: period type of PeriodType enumeration
-    Returns:
-        file index in a pandas.DataFrame with sets of parameters and station id
-    """
-    parameter_path = build_path_to_parameter(parameter, time_resolution, period_type)
-
-    files_server = list_files_of_climate_observations(parameter_path, recursive=True)
 
-    files_server = pd.DataFrame(
-        files_server, columns=[DWDMetaColumns.FILENAME.value], dtype="str"
+    file_index = _create_file_index_for_dwd_server(
+        parameter,
+        time_resolution,
+        period_type,
+        DWDCDCDataPath.CLIMATE_OBSERVATIONS
     )
 
-    # Filter for .zip files
-    files_server = files_server[files_server.FILENAME.str.endswith(ARCHIVE_FORMAT)]
+    file_index = file_index[file_index[DWDMetaColumns.FILENAME.value].str.startswith(ArchiveFormat.ZIP.value)]
 
-    files_server[DWDMetaColumns.FILENAME.value] = files_server[
-        DWDMetaColumns.FILENAME.value
-    ].str.replace(f"{DWD_CDC_PATH}/{DWD_CLIM_OBS_GERMANY_PATH}/", "")
-
-    files_server[DWDMetaColumns.STATION_ID.value] = files_server[
+    file_index[DWDMetaColumns.STATION_ID.value] = file_index[
         DWDMetaColumns.FILENAME.value
     ].apply(lambda x: re.findall(STATION_ID_REGEX, x)[0])
 
-    files_server.loc[:, DWDMetaColumns.STATION_ID.value] = files_server.loc[
+    file_index.loc[:, DWDMetaColumns.STATION_ID.value] = file_index.loc[
         :, DWDMetaColumns.STATION_ID.value
     ].astype(int)
 
-    files_server = files_server.sort_values(
+    file_index = file_index.sort_values(
```

Comment:
Combine this and the last line to:
```
return file_index.sort_values(by=[DWDMetaColumns.STATION_ID.value, DWDMetaColumns.FILENAME.value]
    ).loc[
        :, [DWDMetaColumns.STATION_ID.value, DWDMetaColumns.FILENAME.value]	        :, [DWDMetaColumns.STATION_ID.value, DWDMetaColumns.FILENAME.value]
    ]
```

Reply:
Are you not happy with this comment?

Reply:
I was wondering if such big statement would only lead to confusion for anyone reading it. What do you think?